client/src/services/informationManager.ts

```diff
@@ -0,0 +1,155 @@
+}
+
+export interface IInformation {
+  message: string;
+  level: InformationLevel;
+  data?: object
+}
+
+export class Information {
+  id: string;
+  message: string;
+  level: InformationLevel;
+  data?: object;
+  constructor({ message, level, data = {} }: IInformation) {
+    this.id = uuid4();
+    this.message = message;
+    this.level = level;
+    this.data = data;
+  }
+
+  get theme(): MsReportTheme {
+    switch (this.level) {
+      case InformationLevel.Info:
+        return MsReportTheme.Info;
+      case InformationLevel.Success:
+        return MsReportTheme.Success;
+      case InformationLevel.Warning:
+        return MsReportTheme.Warning;
+      case InformationLevel.Error:
+        return MsReportTheme.Error;
+    }
+  }
+}
+
+export interface InformationOptions {
+  trace?: boolean;
+}
+
+export enum InformationPresentationMode {
+  Toast,
+  Modal,
+  Notification,
+  ToastAndNotification,
+  ModalAndNotification,
+}
+
+export interface InformationToastOptions extends InformationOptions {
+  duration?: number;
+}
+
+export class InformationManager {
+  toastManager: ToastManager;
+  notificationManager: NotificationManager;
+
+  constructor() {
+    this.toastManager = new ToastManager();
+    this.notificationManager = new NotificationManager();
+  }
+
+  async showModal(information: Information): Promise<void> {
+    const alertModalConfig: MsAlertModalConfig = {
+      theme: information.theme,
+      message: information.message,
+    };
+
+    await this._createAndPresentModal(alertModalConfig);
+  }
+
+  async _createAndPresentModal(modalConfig: MsAlertModalConfig): Promise<any> {
+    const top = await modalController.getTop();
+    if (top) {
+      top.classList.add('overlapped-modal');
+    }
+
+    const modal = await modalController.create({
+      component: MsAlertModal,
+      canDismiss: true,
+      cssClass: 'information-modal',
+      componentProps: modalConfig,
+    });
+    await modal.present();
+    const result = await modal.onWillDismiss();
+    await modal.dismiss();
+
+    if (top) {
+      top.classList.remove('overlapped-modal');
+    }
+    return result;
+  }
+
+  async showToast(information: Information, options?: InformationToastOptions): Promise<void> {
+    this.toastManager
+      .createAndPresent({
+        theme: information.theme,
+        message: information.message,
+        duration: options?.duration,
+      })
+      .then(async () => {});
+  }
+
+  sendNotification(information: Information): void {
+    this.notificationManager.create(information);
+  }
+
+  async present(
+    information: Information,
+    presentationMode: InformationPresentationMode = InformationPresentationMode.Toast,
+    options: InformationOptions,
+  ): Promise<void> {
+    if (options && options.trace) {
+      this._trace(information);
+    }
+    switch(presentationMode) {
+      case InformationPresentationMode.Toast:
+        return await this.showToast(information, options);
+        break;
```

Comment:
Return makes the `break` redundant, I'm surprised the linter doesn't cry.

Reply:
copy/paste my bad 🥷 
good catch